api/base/serializers.py

```diff
@@ -88,63 +87,102 @@ def to_internal_value(self, data):
         return super(JSONAPIListField, self).to_internal_value(data)
 
 
-class JSONAPIHyperlinkedIdentityField(ser.HyperlinkedIdentityField):
+class RelationshipField(ser.HyperlinkedIdentityField):
     """
-    HyperlinkedIdentityField that returns a nested dict with url,
-    optional meta information, and link_type.
+    RelationshipField that permits the return of both self and related links, along with optional
+    meta information.
 
     Example:
-
-        children = JSONAPIHyperlinkedIdentityField(view_name='nodes:node-children', lookup_field='pk',
-                                    link_type='related', lookup_url_kwarg='node_id', meta={'count': 'get_node_count'})
-
+    children = RelationshipField(
+        related_view='nodes:node-children',
+        related_view_kwargs={'node_id': 'pk'},
+        self_view='nodes:node-node-children-relationship',
+        self_view_kwargs={'node_id': 'pk'},
+        related_meta={'count': 'get_node_count'}
+    )
     """
+    def __init__(self, **kwargs):
+        related_view = kwargs.pop('related_view', None)
+        self_view = kwargs.pop('self_view', None)
+        related_kwargs = kwargs.pop('related_view_kwargs', None)
+        self_kwargs = kwargs.pop('self_view_kwargs', None)
+        self.views = {'related': related_view, 'self': self_view}
+        self.view_kwargs = {'related': related_kwargs, 'self': self_kwargs}
+        self.related_meta = kwargs.pop('related_meta', None)
+        self.self_meta = kwargs.pop('self_meta', None)
+        assert (related_view is not None or self_view is not None), 'Self or related view must be specified.'
+        if related_view:
+            assert related_kwargs is not None, 'Must provide related view kwargs.'
+        if self_view:
+            assert self_kwargs is not None, 'Must provide self view kwargs.'
+
+        super(RelationshipField, self).__init__(self.views, **kwargs)
+
+    # For retrieving meta values, otherwise returns {}
+    def get_meta_information(self, meta_data, value):
+        meta = {}
+        if meta_data:
+            for key in meta_data or {}:
+                if key == 'count':
+                    show_related_counts = self.context['request'].query_params.get('related_counts', False)
+                    if utils.is_truthy(show_related_counts):
+                        meta[key] = _rapply(meta_data[key], _url_val, obj=value, serializer=self.parent)
+                    elif utils.is_falsy(show_related_counts):
+                        continue
+                    else:
+                        raise InvalidQueryStringError(
+                            detail="Acceptable values for the related_counts query param are 'true' or 'false'; got '{0}'".format(show_related_counts),
```

Comment:
Based on the is_truthy -- is_falsy above this error message isn't __technically__ true but it's better to tell them a good lie and let them do it right the next time.